### PR TITLE
Issue-2-user-can-view-their-entries | Richard Watanabe

### PR DESCRIPTION
**Pull request by Richard-Watanabe**
https://user-images.githubusercontent.com/85139853/125670795-589cd08c-851d-45b7-a270-2d04f0c7c410.mp4

### CODE REVIEW

File: js/main.js [hunk 1]
```diff
@@ -22,8 +22,81 @@ function saveEntry(event) {
   dataObject.entryId = data.nextEntryId;
   data.nextEntryId++;
   data.entries.unshift(dataObject);
-  image.setAttribute('src', 'images/placeholder-image-square.jpg');
+  $image.setAttribute('src', 'images/placeholder-image-square.jpg');
   $form.reset();
+  window.location.reload();
```
Comment (codingcodymiller) [inline js/main.js]: Oh nooooo 😱 

`location.reload()` should never be used, this reloads your page entirely, literally the most inefficient action you could take. If this is to make the new entry show up when you navigate to the entries view, the real answer is to generate a DOM tree for that entry in your `submit` event handler, then `prepend` that entry to the entries list before changing views. You will get the same effect without needing to reload the entire page..

File: js/main.js [hunk 2]
```diff
@@ -22,8 +22,81 @@ function saveEntry(event) {
...
+function goToEntries(event) {
+  $viewEntries.className = 'view';
+  $newModal.className = 'hidden container new';
+}
+
+$entriesButton.addEventListener('click', goToEntries);
+
+var $noEntry = document.querySelector('.noEntry');
+
+if (data.nextEntryId > 1) {
```
Comment (codingcodymiller) [inline js/main.js]: This condition would be better off checking if `data.entries` is empty, you shouldn't use ids as a count.

File: js/main.js [hunk 3]
```diff
@@ -22,8 +22,81 @@ function saveEntry(event) {
...
+}
+
+$newButton.addEventListener('click', popUpNew);
+
+var $entriesButton = document.querySelector('.entries');
+
+function goToEntries(event) {
+  $viewEntries.className = 'view';
+  $newModal.className = 'hidden container new';
+}
```
Comment (codingcodymiller) [inline js/main.js]: You would likely benefit from making one function named something like `switchView` that receives the name of a view as an argument, then shows that view and hides all other views.

**Review by codingcodymiller — CHANGES_REQUESTED** [review]

**Comment by Richard-Watanabe** [comment]
https://user-images.githubusercontent.com/85139853/125696523-3ce6d010-30e0-43f4-85cc-b06240703bb7.mp4

File: js/main.js [hunk 4]
```diff
@@ -22,8 +22,81 @@ function saveEntry(event) {
...
+
+function switchView(event) {
+  if (event.target.matches('.entries-link')) {
+    $viewEntries.className = 'view';
+    $newModal.className = 'container hidden new';
+  } else if (event.target.matches('.new-button')) {
+    $viewEntries.className = 'view hidden';
+    $newModal.className = 'container new';
+  }
+}
```
Comment (codingcodymiller) [inline js/main.js]: You should switch this to a function that takes the name of a view as an argument, then shows that view and hides all other views. That way you don't need to add a new conditional block here for every view you create. I recommend going back to `javascript-view-swapping` for a refresher on how to accomplish this.

File: js/main.js [hunk 5]
```diff
@@ -22,8 +22,81 @@ function saveEntry(event) {
   dataObject.entryId = data.nextEntryId;
   data.nextEntryId++;
   data.entries.unshift(dataObject);
-  image.setAttribute('src', 'images/placeholder-image-square.jpg');
+  $image.setAttribute('src', 'images/placeholder-image-square.jpg');
+  var addNew = createDom(dataObject);
+  $parent.prepend(addNew);
+  $viewEntries.className = 'view';
+  $newModal.className = 'container hidden new';
+  $noEntry.className = 'hidden';
```
Comment (codingcodymiller) [inline js/main.js]: You should handle this with your `switchView` function once you change it to accept an argument of the name of the view to show

**Review by codingcodymiller — CHANGES_REQUESTED** [review]

**Comment by Richard-Watanabe** [comment]
https://user-images.githubusercontent.com/85139853/125834400-24e19653-4f8e-49b6-9720-0e148a000ad9.mp4

File: js/main.js [hunk 6]
```diff
@@ -13,6 +13,7 @@ function uploadImage(event) {
 $imageUrlElement.addEventListener('input', uploadImage);
 
 function saveEntry(event) {
+  switchView(data.view);
```
Comment (codingcodymiller) [inline js/main.js]: this can likely be removed

File: js/main.js [hunk 7]
```diff
@@ -80,23 +80,38 @@ function addEntry(event) {
...
 }
 
-$body.addEventListener('click', switchView);
+function linkSwitch(event) {
+  if (event.target.matches('.link')) {
+    if (event.target.getAttribute('data-view') === 'entries') {
+      switchView('entries');
+    } else {
+      switchView('entry-form');
+    }
```
Comment (codingcodymiller) [inline js/main.js]: You can just save the value of `event.target.getAttribute('data-view')` into a variable and pass that as an argument to `switchView`, which should avoid the need to have a conditional here

**Review by codingcodymiller — CHANGES_REQUESTED** [review]

**Comment by Richard-Watanabe** [comment]
https://user-images.githubusercontent.com/85139853/125836464-d2a55042-d0e3-46fc-aa19-0d18d10bef88.mp4

**Review by codingcodymiller — APPROVED** [review]

**Pull request merged**